githooks/post-checkout

```diff
@@ -5,3 +5,12 @@ if [ "$1" != "$2" ]; then # previous ref != new ref.
   git clean -fd "pkg/**.pb.go" "pkg/**.pb.gw.go" "pkg/**.eg.go" "pkg/**.og.go" "pkg/**generated.go"
   exec git submodule update --init --recursive
 fi
+
+# Only if this is a branch checkout.
+if [ "$3" == "1" ]; then 
+    BRANCH_NAME=$(git symbolic-ref --short -q HEAD)
+    if [ ${BRANCH_NAME} == "master" ]; then
+        git clean -fdx "vendor"
```

Comment:
Nit: I think `rm -rf vendor` accomplishes more or less the same thing here.

Reply:
that's dependent on being in the root directory of the repo, whereas i believe `git clean [pattern]` does not